frontend/src/tabs/segment.pug

```diff
@@ -1,10 +1,10 @@
 .segment(v-bind:class="{ untouched: !segment.authored, active: segment.lines.length < 5 }")
-  .closer(v-if="!segment.authored && segment.lines.length > 4", v-on:click="loadCode()")
+  .closer(v-if="!segment.authored && segment.lines.length > 4", v-on:click="loadCode()" ref="topButton")
     i.fas.fa-plus-circle.icon.open(v-bind:title="'Click to open code'")
     i.fas.fa-chevron-circle-down.icon.hide(v-bind:title="'Click to hide code'")
   div(v-if="loaded", v-hljs="path")
     .code(v-for="(line, index) in segment.lines")
       .line-number {{ segment.lineNumbers[index] + "\n" }}
       .line-content {{ line + "\n" }}
-  .closer.bottom(v-if="!segment.authored && segment.lines.length > 4", v-on:click="loadCode()")
+  .closer.bottom(v-if="!segment.authored && segment.lines.length > 4", v-on:click="collapseCode")
```

Comment:
this change is not needed right?

Reply:
Not absolutely needed, but i separated it so that its less confusing.
e.g. if both are combined, it would be
```
loadCode() {
  this.isLoaded = true

  const segmentTop = this.$refs.topButton.getBoundingClientRect().top;
  if (segmentTop < 0) {
    this.$refs.topButton.scrollIntoView();
  }

  this.nextTick(() => {
    window.toggleNext(this.$el.childNodes[0]);
  }
},
```

It becomes ambiguous which action actually needs what 
Expanding it requires `nextTick` and `this.isLoaded = true`,
but not the scrolling to the top.

Collapsing it conversely dosen't require `nextTick` as the dom is already loaded, nor `this.isLoaded = true`